### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by pyhys**
* Add CHANGELOG.md
* Enable PDF generation
* Describe some LWIP settings
* Improved tutorial and implementation description
* Better in-code documentation
* Logging of incoming and outgoing start-up Ethernet frames

### CODE REVIEW

File: src/device/pf_cmina.c [hunk 1]
```diff
@@ -762,6 +771,20 @@ static const char *pf_cmina_state_to_string(
...
 }
 
+/**
+ * @internal
+ * Print an IPv4 address (without newline)
+ *
+ * @param ip      In: IP address
+*/
+void pf_ip_address_show(uint32_t ip){
+   printf("%"PRIxLEAST32".%"PRIxLEAST32".%"PRIxLEAST32".%"PRIxLEAST32,
```
Comment (hefloryd) [inline src/device/pf_cmina.c:781]: Wouldn't PRIx8 be more correct here?

Comment (pyhys) [reply src/device/pf_cmina.c:781]: ip is a 32-bit integer. I'm getting this compilation error for rt-kernel:
```
/home/jonas/projects/profinetstack/p-net/src/device/pf_cmina.c:781:11: error: format '%hhx' expects argument of type 'int', but argument 2 has type 'long unsigned int' [-Werror=format=]
    printf("%"PRIx8".%"PRIx8".%"PRIx8".%"PRIx8,
           ^~~
       ((ip >> 24) & 0xFF),
       ~~~~~~~~~~~~~~~~~~~

```

File: doc/writing_documentation.rst [hunk 2]
```diff
@@ -9,10 +9,12 @@ Install Sphinx::
     pip3 install -U sphinx-rtd-theme
     pip3 install -U breathe
     pip3 install -U sphinxcontrib-spelling
+    pip3 install -U rst2pdf
```
[the remark this answers is not in the document]
Comment (pyhys) [reply doc/writing_documentation.rst]: Implemented doc/doc_requirements.txt

File: doc/using_codesys.rst [hunk 3]
```diff
@@ -192,3 +214,62 @@ the topmost row. The states of these boolean fields are shown:
 
 If there is no connection at all to the IO-device, the state will shift to
 xBusy from xError once every 5 seconds.
+
+
+Change IP address of IO-device
+------------------------------
+Change the IP address by double-click the "rt_labs_DEMO_device" node in
```
[the remark this answers is not in the document]
Comment (pyhys) [reply doc/using_codesys.rst]: Done

File: doc/getting_started_rtkernel.rst [hunk 4]
```diff
@@ -166,7 +171,39 @@ Use::
     user@host:~/build$ make all
 
 
+PLC timing settings
```
[the remark this answers is not in the document]
Comment (pyhys) [reply doc/getting_started_rtkernel.rst:173]: Reformulated

File: doc/getting_started_rtkernel.rst [hunk 5]
```diff
@@ -153,6 +154,10 @@ Adjust log level
 ----------------
 See the "Getting started on Linux" page.
 
+However note that printing out log strings consume a considerate amount of CPU
```
[the remark this answers is not in the document]
Comment (pyhys) [reply doc/getting_started_rtkernel.rst]: Reformulated

File: doc/getting_started_rtkernel.rst [hunk 6]
```diff
@@ -106,7 +106,8 @@ Use the menu "File > Import". Select "General > Existing Projects". Click
 
 Run on target
 -------------
-First start Segger J-link GDB debug server::
+Install J-link for Linux from https://www.segger.com/
```
[the remark this answers is not in the document]
Comment (pyhys) [reply doc/getting_started_rtkernel.rst]: Done

File: README.md [hunk 7]
```diff
@@ -69,6 +70,12 @@ For rt-kernel:
 
 * Workbench 2017.1 or later
 
+An example of microcontroller we have been using is the Infineon XMC4800,
+which has an ARM Cortex-M4 running at 144 MHz, with 2 MB Flash and 352 kB RAM.
+It runs the rt-kernel RTOS, and we have tested it with 9 Profinet slots each
```
[the remark this answers is not in the document]
Comment (pyhys) [reply README.md]: Done

File: README.md [hunk 8]
```diff
@@ -18,12 +18,14 @@ implementations. It is easy to use and provides a small footprint. It
 is especially well suited for embedded systems where resources are
 limited and efficiency is crucial.
 
-It is written in C and can be run on bare-metal hardware, an RTOS such
-as rt-kernel, or on Linux. The main requirement is that the
+It is written in C (version C99) and can be run on bare-metal hardware, an
```
[the remark this answers is not in the document]
Comment (pyhys) [reply README.md]: Moved to "Implementation details" page.

File: src/device/pf_cmina.c (hunk 1 above)
Comment (hefloryd) [reply src/device/pf_cmina.c:781]: Hmm, that's unexpected. I tested with a generic arm-none-eabi-gcc and see the same issue, but it works for x64-gcc. Not sure why. PRIx32 works and is maybe a bit cleaner. Or cast to uint8_t (and maybe avoid inttypes.h), like we do in most other places.

File: README.md [hunk 9]
```diff
@@ -18,12 +18,14 @@ implementations. It is easy to use and provides a small footprint. It
...
 
-It is written in C and can be run on bare-metal hardware, an RTOS such
-as rt-kernel, or on Linux. The main requirement is that the
+It is written in C and can be run on bare-metal hardware, an RTOS such as
+rt-kernel, or on Linux. The main requirement is that the
 platform can send and receive raw Ethernet Layer 2 frames. The
 p-net stack is supplied with full sources including a porting
 layer.
 
+Also C++ (version C++11 or later) is supported.
```
[the remark this answers is not in the document]
Comment (pyhys) [reply README.md]: Moved to "Implementation details" page. On README.md: 
`Also C++ (any version) is supported.`

File: doc/getting_started_rtkernel.rst [hunk 10]
```diff
@@ -153,6 +154,9 @@ Adjust log level
 ----------------
 See the "Getting started on Linux" page.
 
+However note that printing out log strings is very slow, so you probably need
```
[the remark this answers is not in the document]
Comment (pyhys) [reply doc/getting_started_rtkernel.rst]: Changed to:
`In order to learn the Profinet communication model, it is very informative to adjust the log level to see the incoming and outgoing messages. See the "Getting started on Linux" page for details on how to adjust the log level.

However note that printing out log strings is slow, so you probably need to decrease the cyclic data frequency (see PLC timing settings below).`

File: doc/getting_started_rtkernel.rst [hunk 11]
```diff
@@ -166,7 +170,42 @@ Use::
...
+
 Memory requirements for the tests
 ---------------------------------
 Note that the tests require a stack of at least 6 kB. You may have to increase
 CFG_MAIN_STACK_SIZE in your BSP ``include/config.h`` file.
+
+
+IP-stack lwip
+-------------
+The rt-kernel uses the "lwip" IP stack (version 2.0.2 at the time of writing).
```
[the remark this answers is not in the document]
Comment (pyhys) [reply doc/getting_started_rtkernel.rst]: Removed

File: doc/writing_documentation.rst [hunk 12]
```diff
@@ -4,15 +4,12 @@ Installing documentation toolchain
 Install Sphinx::
 
     sudo apt install python3-pip
```
[the remark this answers is not in the document]
Comment (pyhys) [reply doc/writing_documentation.rst]: Done

File: src/device/pf_cmina.c (hunk 1 above)
Comment (pyhys) [reply src/device/pf_cmina.c]: Did cast to unsigned

**Pull request merged**